### PR TITLE
Upgrade to Cadence 1.0

### LINKED ISSUE #12: Upgrade code to Cadence 1
### Issue To Be Solved
Contract and transactions code need to be updated to C1

### PR DESCRIPTION
**Pull request by franklywatson**
First pass attempt at Cadence 1.0 upgrade. Did not run the Cadence tests yet but wanted to get a sanity check first

### CODE REVIEW

File: contracts/BandExampleConsumerToken.cdc [hunk 1]
```diff
@@ -27,45 +27,45 @@ event emitted by the BandOracle contract. When a symbol used by the dapp has it'
...
     // calculated by getting the FLOW/USD rate from the band oracle
-    pub var tokenUSDPrice: UFix64
-    pub var tokenFlowPrice: UFix64
+    access(all) var tokenUSDPrice: UFix64
+    access(all) var tokenFlowPrice: UFix64
     
     // The funds collecting minting tokens will be stored here. This vault will be 
     // also used to pay for the oracle fees.
-    pub let flowTreasure: @FungibleToken.Vault
+    access(all) let flowTreasure: @FungibleToken.Vault
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc]:
```suggestion
    access(self) let flowTreasure: @FungibleToken.Vault
```

File: contracts/BandExampleConsumerToken.cdc [hunk 2]
```diff
@@ -98,7 +98,7 @@ pub contract BandExampleConsumerToken: FungibleToken {
         // created Vault to the context that called so it can be deposited
         // elsewhere.
         //
-        pub fun withdraw(amount: UFix64): @FungibleToken.Vault {
+        access(all) fun withdraw(amount: UFix64): @FungibleToken.Vault {
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc]:
```suggestion
        access(FungibleToken.Withdraw) fun withdraw(amount: UFix64): @{FungibleToken.Vault} {
```

File: contracts/BandExampleConsumerToken.cdc [hunk 3]
```diff
@@ -79,10 +79,10 @@ pub contract BandExampleConsumerToken: FungibleToken {
     // out of thin air. A special Minter resource needs to be defined to mint
     // new tokens.
     //
-    pub resource Vault: FungibleToken.Provider, FungibleToken.Receiver, FungibleToken.Balance {
+    access(all) resource Vault: FungibleToken.Provider, FungibleToken.Receiver, FungibleToken.Balance {
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc]:
```suggestion
    access(all) resource Vault: FungibleToken.Vault {
```

File: contracts/BandExampleConsumerToken.cdc [hunk 4]
```diff
@@ -133,19 +131,19 @@ pub contract BandExampleConsumerToken: FungibleToken {
     // and store the returned Vault in their storage in order to allow their
     // account to be able to receive deposits of this token type.
     //
-    pub fun createEmptyVault(): @FungibleToken.Vault {
+    access(all) fun createEmptyVault(): @FungibleToken.Vault {
         return <-create Vault(balance: 0.0)
     }
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc:139]: Adding new interface conformance methods
```suggestion

        /// Called when a fungible token is burned via the `Burner.burn()` method
        access(contract) fun burnCallback() {
            if self.balance > 0.0 {
                BandExampleConsumerToken.totalSupply = BandExampleConsumerToken.totalSupply - self.balance
            }
            self.balance = 0.0
        }

        access(all) view fun getViews(): [Type] {
            return BandExampleConsumerToken.getContractViews(resourceType: nil)
        }

        access(all) fun resolveView(_ view: Type): AnyStruct? {
            return BandExampleConsumerToken.resolveContractView(resourceType: nil, viewType: view)
        }

        /// getSupportedVaultTypes optionally returns a list of vault types that this receiver accepts
        access(all) view fun getSupportedVaultTypes(): {Type: Bool} {
            let supportedTypes: {Type: Bool} = {}
            supportedTypes[self.getType()] = true
            return supportedTypes
        }

        access(all) view fun isSupportedVaultType(type: Type): Bool {
            return self.getSupportedVaultTypes()[type] ?? false
        }

        /// Asks if the amount can be withdrawn from this vault
        access(all) view fun isAvailableToWithdraw(amount: UFix64): Bool {
            return amount <= self.balance
        }
```

File: contracts/BandExampleConsumerToken.cdc [hunk 5]
```diff
@@ -27,45 +27,45 @@ event emitted by the BandOracle contract. When a symbol used by the dapp has it'
 transaction should be run to update price quotes within the contract. In the current example calling the
 `updateTokenFlowPrice()` method from the admin resource achieves this.
 **/
-pub contract BandExampleConsumerToken: FungibleToken {
+access(all) contract BandExampleConsumerToken: FungibleToken {
 
     // The USD price is determined by the admin, then the equivalent Flow price is
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc:34]: Adding an entitlement for use in the `Administrator` resource.

```suggestion
    access(all) entitlement Update
    
    // The USD price is determined by the admin, then the equivalent Flow price is 
```

File: contracts/BandExampleConsumerToken.cdc [hunk 6]
```diff
@@ -133,19 +131,19 @@ pub contract BandExampleConsumerToken: FungibleToken {
...
+    access(all) fun createEmptyVault(): @FungibleToken.Vault {
         return <-create Vault(balance: 0.0)
     }
 
     // Admin resource that allows to set the token price
-    pub resource Administrator {
+    access(all) resource Administrator {
         
-        pub fun setNewTokensUSDPrice (price: UFix64) {
+        access(all) fun setNewTokensUSDPrice (price: UFix64) {
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc]: Implementing the above entitlement. Not likely that this resource will be linked publicly, but good practice with downcasting now available.

```suggestion
        access(Update) fun setNewTokensUSDPrice (price: UFix64) {
```

File: contracts/BandExampleConsumerToken.cdc [hunk 7]
```diff
@@ -133,19 +131,19 @@ pub contract BandExampleConsumerToken: FungibleToken {
...
             BandExampleConsumerToken.tokenUSDPrice = price
             BandExampleConsumerToken.updateTokenFlowPrice()
         }
 
-        pub fun updateTokenFlowPrice () {
+        access(all) fun updateTokenFlowPrice () {
```
Comment (sisyphusSmiling) [inline contracts/BandExampleConsumerToken.cdc]: Same as above

```suggestion
        access(Update) fun updateTokenFlowPrice () {
```

File: contracts/BandOracle.cdc [hunk 8]
```diff
@@ -4,20 +4,20 @@ import "FlowToken"
...
-pub contract BandOracle {
+access(all) contract BandOracle {
     
     /// Paths
 
     // OracleAdmin resource paths.
-    pub let OracleAdminStoragePath: StoragePath
-    pub let OracleAdminPrivatePath: PrivatePath
+    access(all) let OracleAdminStoragePath: StoragePath
+    access(all) let OracleAdminPrivatePath: PrivatePath
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]: We can remove these `PrivatePath` fields now

File: contracts/BandOracle.cdc [hunk 9]
```diff
@@ -94,32 +94,32 @@ pub contract BandOracle {
 
     /// Admin only operations.
     ///
-    pub resource interface OracleAdmin {
-        pub fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath
-        pub fun removeSymbol (symbol: String)
-        pub fun createNewFeeCollector (): @BandOracle.FeeCollector
+    access(all) resource interface OracleAdmin {
+        access(all) fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc:105]: This method can probably just be removed. We'll have to think about how to handle Capabilities here 🤔

File: contracts/BandOracle.cdc [hunk 10]
```diff
@@ -189,7 +189,7 @@ pub contract BandOracle {
         /// @param resolveTime: The registered time for the rates.
         /// @param requestID: The Band Protocol request ID.
         ///
-        pub fun relayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
+        access(all) fun relayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun relayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
```

File: contracts/BandOracle.cdc [hunk 11]
```diff
@@ -201,7 +201,7 @@ pub contract BandOracle {
         /// @param resolveTime: The registered time for the rates.
         /// @param requestID: The Band Protocol request ID.
         ///
-        pub fun forceRelayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
+        access(all) fun forceRelayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun forceRelayRates (symbolsRates: {String: UInt64}, resolveTime: UInt64, requestID: UInt64) {
```

File: contracts/BandOracle.cdc [hunk 12]
```diff
@@ -170,15 +170,15 @@ pub contract BandOracle {
...
 
     }
 
     /// The resource that will allow an account to make quote updates
     ///
-    pub resource Relay {
+    access(all) resource Relay {
         
         // Capability linked to the OracleAdmin allowing relayers to relay rate updates
         access(self) let updaterCapability: Capability<&{DataUpdater}>
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(self) let updaterCapability: Capability<auth(Update) &{DataUpdater}>
```

File: contracts/BandOracle.cdc [hunk 13]
```diff
@@ -170,15 +170,15 @@ pub contract BandOracle {
         ///
         /// @return The `FeeCollector` resource
         ///
-        pub fun createNewFeeCollector (): @FeeCollector {
+        access(all) fun createNewFeeCollector (): @FeeCollector {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Fee) fun createNewFeeCollector (): @FeeCollector {
```

File: contracts/BandOracle.cdc [hunk 14]
```diff
@@ -146,7 +146,7 @@ pub contract BandOracle {
         /// @param requestID: The Band Protocol request ID.
         /// @param relayerID: The ID of the relayer carrying the update.
         ///
-        pub fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+        access(all) fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```

File: contracts/BandOracle.cdc [hunk 15]
```diff
@@ -159,7 +159,7 @@ pub contract BandOracle {
         /// @param requestID: The Band Protocol request ID.
         /// @param relayerID: The ID of the relayer carrying the update.
         ///
-        pub fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+        access(all) fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```

File: contracts/BandOracle.cdc [hunk 16]
```diff
@@ -135,7 +135,7 @@ pub contract BandOracle {
         ///
         /// @param symbol: The string representing the symbol to be removed from the contract.
         ///
-        pub fun removeSymbol (symbol: String) {
+        access(all) fun removeSymbol (symbol: String) {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Remove) fun removeSymbol (symbol: String) {
```

File: contracts/BandOracle.cdc [hunk 17]
```diff
@@ -94,32 +94,32 @@ pub contract BandOracle {
...
+        access(all) fun removeSymbol (symbol: String)
+        access(all) fun createNewFeeCollector (): @BandOracle.FeeCollector
     }
 
     /// Relayer operations.
     ///
-    pub resource interface DataUpdater {
-        pub fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+    access(all) resource interface DataUpdater {
+        access(all) fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```

File: contracts/BandOracle.cdc [hunk 18]
```diff
@@ -94,32 +94,32 @@ pub contract BandOracle {
...
                             requestID: UInt64, relayerID: UInt64)
-        pub fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+        access(all) fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Update) fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64,
```

File: contracts/BandOracle.cdc [hunk 19]
```diff
@@ -94,32 +94,32 @@ pub contract BandOracle {
 
     /// Admin only operations.
     ///
-    pub resource interface OracleAdmin {
-        pub fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath
-        pub fun removeSymbol (symbol: String)
-        pub fun createNewFeeCollector (): @BandOracle.FeeCollector
+    access(all) resource interface OracleAdmin {
+        access(all) fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath
+        access(all) fun removeSymbol (symbol: String)
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Remove) fun removeSymbol (symbol: String)
```

File: contracts/BandOracle.cdc [hunk 20]
```diff
@@ -94,32 +94,32 @@ pub contract BandOracle {
...
+        access(all) fun createNewFeeCollector (): @BandOracle.FeeCollector
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Fee) fun createNewFeeCollector (): @BandOracle.FeeCollector
```

File: contracts/BandOracle.cdc [hunk 21]
```diff
@@ -216,21 +216,21 @@ pub contract BandOracle {
...
     ///
-    pub resource FeeCollector {
+    access(all) resource FeeCollector {
 
         /// Sets the fee in Flow tokens for the oracle use.
         ///
         /// @param fee: The amount of Flow tokens.
         ///
-        pub fun setFee (fee: UFix64) {
+        access(all) fun setFee (fee: UFix64) {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Fee) fun setFee (fee: UFix64) {
```

File: contracts/BandOracle.cdc [hunk 22]
```diff
@@ -216,21 +216,21 @@ pub contract BandOracle {
...
             BandOracle.setFee(fee: fee)
         }
 
         /// Extracts the fees from the contract's vault.
         /// 
         /// @return A vault containing the funds obtained for the oracle use.
         ///
-        pub fun collectFees (): @FungibleToken.Vault {
+        access(all) fun collectFees (): @FungibleToken.Vault {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        access(Fee) fun collectFees (): @FungibleToken.Vault {
```

File: contracts/BandOracle.cdc [hunk 23]
```diff
@@ -344,7 +344,7 @@ pub contract BandOracle {
     /// @param updaterCapability: The capability pointing to the OracleAdmin resource needed to create the relay.
     /// @return The new relay resource.
     ///
-    pub fun createRelay (updaterCapability: Capability<&{DataUpdater}>): @Relay {
+    access(all) fun createRelay (updaterCapability: Capability<&{DataUpdater}>): @Relay {
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
    access(all) fun createRelay (updaterCapability: Capability<auth(Update) &{DataUpdater}>): @Relay {
```

File: contracts/BandOracle.cdc [hunk 24]
```diff
@@ -425,7 +425,8 @@ pub contract BandOracle {
         self.FeeCollectorStoragePath = /storage/BandOracleFeeCollector
         self.dataUpdaterPrivateBasePath = "BandOracleDataUpdater"
         self.account.save(<- create BandOracleAdmin(), to: self.OracleAdminStoragePath)
```
Comment (sisyphusSmiling) [inline contracts/BandOracle.cdc]:
```suggestion
        self.account.storage.save(<- create BandOracleAdmin(), to: self.OracleAdminStoragePath)
```

**Review by sisyphusSmiling — COMMENTED** [review]
Adding what I have so far. Will follow up with a PR against this one addressing the private Capability handling.

File: contracts/BandOracle.cdc (hunk 8 above)
Comment (franklywatson) [inline contracts/BandOracle.cdc]:
```suggestion
```

File: contracts/BandOracle.cdc [hunk 25]
```diff
@@ -4,20 +4,20 @@ import "FlowToken"
...
-    pub let OracleAdminStoragePath: StoragePath
-    pub let OracleAdminPrivatePath: PrivatePath
+    access(all) let OracleAdminStoragePath: StoragePath
+    access(all) let OracleAdminPrivatePath: PrivatePath
 
     // Relay resource paths.
-    pub let RelayStoragePath: StoragePath
-    pub let RelayPrivatePath: PrivatePath
+    access(all) let RelayStoragePath: StoragePath
+    access(all) let RelayPrivatePath: PrivatePath
```
Comment (franklywatson) [inline contracts/BandOracle.cdc]:
```suggestion
```

File: contracts/utility/FlowToken.cdc [hunk 26]
```diff
@@ -57,9 +75,19 @@ pub contract FlowToken: FungibleToken {
...
         //
-        pub fun withdraw(amount: UFix64): @FungibleToken.Vault {
+        access(FungibleToken.Withdraw) fun withdraw(amount: UFix64): @{FungibleToken.Vault} {
             self.balance = self.balance - amount
-            emit TokensWithdrawn(amount: amount, from: self.owner?.address)
+            if let address = self.owner?.address {
+                if address != 0xf8d6e0586b0a20c7 &&
+                   address != 0xf4527793ee68aede &&
+                   address != 0x9eca2b38b18b5dfe &&
+                   address != 0x8624b52f9ddcd04a
```
Comment (franklywatson) [inline contracts/utility/FlowToken.cdc:84]: What are these hardcoded addresses about? Probably worth a comment

Comment (sisyphusSmiling) [reply contracts/utility/FlowToken.cdc:84]: Looks like they're service accounts across various networks. I think it's related to minimizing the number of events emitted during service account transactions.

File: contracts/BandOracle.cdc [hunk 27]
```diff
@@ -4,21 +4,23 @@ import "FlowToken"
...
+    access(all) let RelayStoragePath: StoragePath
 
     // FeeCollector resource paths.
-    pub let FeeCollectorStoragePath: StoragePath
+    access(all) let FeeCollectorStoragePath: StoragePath
 
+    // Entitlements
+    access(all) entitlement Fees
+    access(all) entitlement Update
+    access(all) entitlement Remove
```
Comment (alilloig) [inline contracts/BandOracle.cdc]: @sisyphusSmiling the language interpreter is marking this entitlement as previously declared, perhaps this one is a generic entitlement of Cadence?

File: contracts/utility/FlowToken.cdc (hunk 26 above)
Comment (alilloig) [reply contracts/utility/FlowToken.cdc:84]: I know the first address is the emulator account for sure

File: contracts/BandOracle.cdc (hunk 27 above)
Comment (alilloig) [reply contracts/BandOracle.cdc]: It seems so, so I'm calling it something slightly different

Captura de pantalla 2024-08-01 a las 16 33 45

**Pull request closed without merging**